qubesbuilder/plugins/installer/__init__.py

```diff
@@ -0,0 +1,515 @@
+
+            # Add prepared chroot cache
+            if chroot_cache.exists():
+                copy_in += [(chroot_cache, Path("/var/cache/mock"))]
+
+            copy_out = [
+                (
+                    self.executor.get_plugins_dir()
+                    / f"installer/work/{self.iso_version}/{self.dist.architecture}/iso/{self.iso_name}.iso",
+                    iso_dir,
+                )
+            ]
+
+            # Prepare installer cmd
+            cmd = []
+
+            # Create builder-local repository (could be empty) inside the cage
+            cmd += [
+                f"mkdir -p {self.executor.get_repository_dir()}",
+                f"cd {self.executor.get_repository_dir()}",
+                "createrepo_c .",
+            ]
+
+            #
+            # Inside mock chroot to use qubes-lorax
+            #
+
+            mock_cmd = [
+                f"sudo --preserve-env={','.join(self.environment.keys())}",
+                f"/usr/libexec/mock/mock",
+                f"--root {self.executor.get_plugins_dir()}/installer/mock/{mock_conf}",
+                "--disablerepo='*'",
+                f"--chroot 'env {self.get_env()} make -C {self.executor.get_plugins_dir()}/installer iso-prepare iso-parse-kickstart iso-installer-lorax iso-installer-mkisofs'",
+            ]
+            if isinstance(self.executor, ContainerExecutor):
+                msg = (
+                    f"{self.dist}: Mock isolation set to 'simple', build has full network "
+                    f"access. Use 'qubes' executor for network-isolated build."
+                )
+                log.warning(msg)
+                mock_cmd.append("--isolation=simple")
+            else:
+                mock_cmd.append("--isolation=nspawn")
+            if self.verbose:
+                mock_cmd.append("--verbose")
+            if chroot_cache.exists():
+                mock_cmd.append("--plugin-option=root_cache:age_check=False")
+
+            files_inside_executor_with_placeholders = [
+                f"{self.executor.get_plugins_dir()}/installer/mock/{mock_conf}"
+            ]
+
+            cmd += [" ".join(mock_cmd)]
+
+            try:
+                self.executor.run(
+                    cmd,
+                    copy_in,
+                    copy_out,
+                    environment=self.environment,
+                    files_inside_executor_with_placeholders=files_inside_executor_with_placeholders,
+                )
+            except ExecutorError as e:
+                msg = f"{self.dist}: Failed to create ISO: {str(e)}."
+                raise InstallerError(msg) from e
+
+            # Save ISO information we built
+            info = {
+                "iso": iso.name,
+                "kickstart": str(self.installer_kickstart),
+                "packages": {
+                    "runtime": [
+                        pkg.name
+                        for pkg in (cache_dir / self.iso_name / "rpm").glob("*.rpm")
```

Comment:
There should be no other files than rpm, right? Maybe that's a use case for `verify` stage, to check if really just rpm files are here, and os/Packages below?

Reply:
Yes there is nothing else than RPM normally. It's a good idea to add such check in verify stage.

Reply:
I think we should even do that check ealier before build.